bindings/rust/s2n-tls/src/raw/config.rs

```diff
@@ -2,70 +2,159 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::raw::{
+    connection::Connection,
     error::{Error, Fallible},
     security,
 };
-use alloc::sync::Arc;
-use core::{convert::TryInto, ptr::NonNull};
+use core::{convert::TryInto, ptr::NonNull, task::Poll};
 use s2n_tls_sys::*;
-use std::ffi::CString;
+use std::{
+    ffi::{c_void, CString},
+    mem::ManuallyDrop,
+    sync::atomic::{AtomicUsize, Ordering},
+};
 
-struct Owned(NonNull<s2n_config>);
+pub struct Config(NonNull<s2n_config>);
 
-/// Safety: s2n_config objects can be sent across threads
-unsafe impl Send for Owned {}
+/// # Safety
+///
+/// Callers must ensure access to Config is thread safe
```

Comment:
This isn't correct. Configs are thread safe as long as the handlers/callbacks are threadsafe, which we will ensure in the trait requirements.

Reply:
I was conflating the Send and thread-safe concepts. This was a pretty good read: https://onesignal.com/blog/thread-safety-rust/